tests/ui/async-await/format-await-send.rs

```diff
@@ -0,0 +1,24 @@
+// regression test for <https://github.com/rust-lang/rust/issues/101650>
+// assert that Future which has format!() with an async function is Send
+
+#![allow(unused)]
+
+//@ check-pass
+//@ edition: 2018
+
+use core::future::Future;
+use core::pin::Pin;
+
+fn build_string() -> Pin<Box<dyn Future<Output = String> + Send>> {
+    Box::pin(async move {
+        let mut string_builder = String::new();
+        string_builder += &format!("Hello {}", helper().await);
+        string_builder
```

Comment:
```suggestion
        format!("Hello {}", helper().await)
```
this should also already test test, does it not?

Reply:
According to that [comment](https://github.com/rust-lang/rust/issues/101650#issuecomment-2795952082), `format!(..)` and `string += &format!(..);` are separate cases. The `format!(..)` case has already been tested in [this file](https://github.com/rust-lang/rust/blob/a43b8d1d2b10bf4d9e9d935aed497f7ebbeeb6e4/tests/ui/fmt/format-with-yield-point.rs).

Also, it might be better to move the test into `tests/ui/fmt` and give it a new name.